### PR TITLE
MES-1431 - Updated Login Screen

### PR DESCRIPTION
**Pull request by Cavalier07**
## Description and relevant Jira numbers

The aim of the PR was to update the login screen so it provides more information. Changes include:

- Moved all authentication logic from App to LoginPage
- Updated Authentication Provider so it passes back a error message
- Updated Login Page to show correct message and offer required actions

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [ ] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [x] PR link added to JIRA
- [x] Reviewers selected in Github
- [x] Any new reusable component/widget added to component library on Confluence

- [x] Screenshot(s) captured on the physical device (if applicable)
- [x] Tested by QA
- [x] PO's approval

### CODE REVIEW

**Comment by Cavalier07** [comment]
User has logged out Login Screen:

unadjustednonraw_thumb_3c

Login Failed due to no Internet Screen:

unadjustednonraw_thumb_3b

Login failed as user cancelled action screen

TODO

Login Failed due to unexpected error screen

TODO

File: package.json [hunk 1]
```diff
@@ -45,9 +45,11 @@
     "@angular/http": "5.2.11",
     "@angular/platform-browser": "5.2.11",
     "@angular/platform-browser-dynamic": "5.2.11",
+    "@ionic-native-mocks/open-native-settings": "^2.0.12",
```
Comment (johnmorleybjss) [inline package.json]: Shouldn't this be in `devDependencies`?

File: src/pages/login/__tests__/login.spec.ts [hunk 2]
```diff
@@ -21,30 +29,149 @@ describe('LoginPage', () => {
...
+
       component.login().then(() => {
         expect(navController.setRoot).toBeCalledWith('JournalPage');
-      })
+        expect(component.hasUserLoggedOut).toBeFalsy();
+        expect(splashScreen.hide).toBeCalledTimes(1);
+      });
+    });
+
+    it('should should not log in succesfully', () => {
```
Comment (johnmorleybjss) [inline src/pages/login/__tests__/login.spec.ts]: The name of this test could be better.

File: package.json (hunk 1 above)
Comment (Cavalier07) [reply package.json]: My bad moved it

File: src/pages/login/__tests__/login.spec.ts (hunk 2 above)
Comment (Cavalier07) [reply src/pages/login/__tests__/login.spec.ts]: Have renamed

**Comment by Cavalier07** [comment]
@ammarhaiderbjss  - Have moved the package to dev dependencies 👍

**Comment by Cavalier07** [comment]
Added new screenshots based on new copy

**Pull request merged**